### PR TITLE
sparkle: Static Data Source screens

### LINKED ISSUE #2201: [Doom] Static Data Sources improvements
From: https://github.com/dust-tt/decisions/issues/105

### Folder metaphore

image

Clarify the notion of a bucket of documents.

### Document management

image

Proposed changes:
- We create a field for Document title
- We regroup tags and document ID behind a "Developer option" setting button (opens a side modal)

### PR DESCRIPTION
**Pull request by spolu**
Fixes: https://github.com/dust-tt/dust/issues/2201

Refactors upsert page and document list page. Follows framing in the issue above with an exception for tags: we only display them if they have been set by API otherwise we hide that section entirely. The design was not perfectly clear and this strikes a decent 80/20.

Screenshot from 2023-10-31 18-24-07
Screenshot from 2023-10-31 18-24-16
Screenshot from 2023-10-31 18-24-27

### CODE REVIEW

File: front/pages/w/[wId]/builder/data-sources/[name]/upsert.tsx [hunk 1]
```diff
@@ -283,213 +284,123 @@ export default function DataSourceUpsert({
...
+        <Page.Vertical align="stretch">
+          <div className="pt-4">
+            <Page.SectionHeader title="Document title" />
+            <div className="pt-4">
+              <Input
+                placeholder="Document title"
+                name="document"
+                disabled={readOnly || !!loadDocumentId}
+                value={documentId}
+                onChange={(v) => setDocumentId(v)}
```
Comment (PopDaph) [inline front/pages/w/[wId]/builder/data-sources/[name]/upsert.tsx:297]: We don't need the target.value?

**Review by PopDaph — DISMISSED** [review]

File: front/pages/w/[wId]/builder/data-sources/[name]/upsert.tsx (hunk 1 above)
Comment (spolu) [reply front/pages/w/[wId]/builder/data-sources/[name]/upsert.tsx:297]: sparkle returns the value directly it seems

**Comment by spolu** [comment]
Actually will make another pass on this one before merging to push it a notch further

@dust-tt/product-team plan for static data sources from UI:
- uniquely generated ID for documentId and standardize on using the tag `tite:xxx` so that it can be edited

Any FUD about this?

**Comment by fontanierh** [comment]
> https://github.com/orgs/dust-tt/teams/product-team plan for static data sources from UI

if you pushed stuff using API then it will be hard to understand the UI no ?

**Comment by spolu** [comment]
Yeah... I agree that there is still something to be figured out there... maybe won't action this yet. Will still preserve tags setting and align better with the figma

**Comment by spolu** [comment]
Added a collapsible-ish section Developer Options close to design with ability to show and edit (keep the edit) tags

**Pull request merged**